general/releases.rst

```diff
@@ -0,0 +1,146 @@
+
+Cadence
+*******
+New versions of OP-TEE are released four times a year, i.e., quarterly releases.
+The releases have historically lined up with Linaro Connect events. I.e., a
+release has been made right after Linaro Connect and one release somewhere
+between two Linaro Connects. I.e., typically there has been releases in January,
+April, July and October.
+
+.. _releases_changelog:
+
+Changelog
+*********
+The changelog is stored in the :ref:`optee_os` git (CHANGELOG.md_). There you
+can see what has been done between the different releases in terms of commits as
+well as pull requests. It's worth noting that this changelog only lists changes
+in optee_os and not the other optee_xyz gits. However, most work is done in that
+git and often changes in optee_os also require changes in :ref:`optee_client`
+and :ref:`optee_test`.
+
+.. _releases_versioning_schema:
+
+Versioning schema
+*****************
+OP-TEE follows `Semantic Versioning 2.0.0`_. What that means in practice is well
+described at the link just shown.
+
+.. _releases_release_procedure:
+
+Release procedure
+*****************
+There are certain steps that needs to be done when making a release. This
+checklist here serves as guidance to the one in charge of making a new release.
+Roughly start with this 2-3 weeks before the targeted release date.
+
+tl;dr
+=====
+.. list-table:: Short version of the OP-TEE release procedure
+    :widths: 60 300 10
+    :header-rows: 1
+
+    * - When
+        (Tminus)
+      - Action
+      - Example
+
+    * - 3w
+      - Create release pull request
+      - `PR#3099`_
+
+    * - 3w
+      - Inform maintainers about upcoming release
+      -
+
+    * - 1w
+      - Create RC-x tag in optee_* + build.git
+      - git tag -a 3.x.y-rc1 -m "3.x.y-rc1"
+
+    * - 1w
+      - Inform maintainers about RC tag
+      -
+
+    * - 1w
+      - Test platform builds / devices
+      - 
+
+    * - Release day
+      - Update CHANGELOG.md_ 
+      - `changelog example`_
+
+    * - Release day
+      - Collect/merge ``Tested-By`` tags
+      - `commit example`_
+
+    * - Release day
+      - Create release branch in :ref:`manifest`
+      - git checkout -b 3.x.y github/master
+
+    * - Release day
+      - Update manifest XML-files
+      - `3.6.0 stable`_
+
+    * - Release day
+      - Create release tag in optee_* + build.git
+      - git tag -a 3.x.y -m "3.x.y"
+
+    * - Release day
+      - Inform maintainers and stakeholder that release has been completed.
+      -
+
+
+Long version
+============
+
+    1. Create a "release pull request" at GitHub ought to collect ``Tested-By``
+       tags from various maintainers. As an example, see `PR#3099`_.
+
+    2. Send email to all maintainers to let them know about the upcoming
+       release.
+       
+    3. Create a release candidate (RC) tag (annotated tag, i.e., ``git tag -a
+       3.x.y-rc1 -m "3.x.y-rc1"``) in the following gits
+       ``optee_{os,client,test,examples,docs}`` and ``build.git``.
```

Comment:
And `optee_benchmark` too. Better use `optee*`?
The bash commands I'm using:
```
export VER=3.x.y-rc1
for d in optee* build; do ( cd $d; git tag -a $VER -m $VER ); done
for d in optee* built; do ( cd $d; git push upstream $VER ); done
```

Reply:
Agree and I'll add the commands as a hint, with a minor change 
```diff
- for d in optee* built; do ( cd $d; git push upstream $VER ); done
+ for d in optee* built; do ( cd $d; git push github $VER ); done
```
since `github` is the name of the remote in our default development setup.

Reply:
> `github` is the name of the remote in our default development setup.

Yes but it's cloned with the public HTTPS URL, not the SSH one (git@github.com:OP-TEE/optee_os.git for instance). So you can't push to it.

Reply:
I thought password worked for HTTPS based cloning, but I could be wrong. Maybe better to just say `origin` and leave it up to the reader to figure out what to do. It's not really rocket science so to say.

Reply:
Well, I could very well be the one who's wrong, because I think I have never tried to push to GitHub over HTTPS ;-) SSH is just so convenient.

Leaving `origin` here is fine.